### PR TITLE
Stop sorting arrays in schema export

### LINKED ISSUE #21391: Schema export sorts order of fields in interfaces
### Describe the Bug

When using the schema export functionality, we noticed that on importing, all our fields on the list-o2m interface were reset to be alphabetically ordered. Upon investigating the `schema.json` file we saw that on a field under `meta.options.fields` the array has been sorted.

Fix in PR

### To Reproduce

Added a relationship field with the list-o2m interface, add fields to the list in a non alphabetical order.
Run `schema export`, you might need to change something manually or in another environment for the hash to update, but once you then run `schema import` you'll notice those fields have now been sorted alphabetically.

### Directus Version

10.9.0

### Hosting Strategy

Self-Hosted (Docker Image)

### PR DESCRIPTION
**Pull request by u12206050**
What's changed:
Stops sorting any arrays in the schema export in order to keep order consistent

Fixes #21391

### CODE REVIEW

**Review by Nitwel — APPROVED** [review]

**Comment by Nitwel** [comment]
if you remove `isArray`, this should be ready to be merged. 👍

**Comment by u12206050** [comment]
Hmm I realise the problem with in sorted object keys is back as it is inconsistent between environments causing useless commit changes. That was part of the original reason for sorting schema, which my original PR maintained by still sorting objects in arrays

**Comment by Nitwel** [comment]
There are several points that don't add up to me with this code in general.
- The top level array are sorted anyways because of the `sortBy` in lines 32-38.
- keys of objects in arrays in deeper levels shouldn't have been sorted at all before either because the `sortDeep` function only gets recursively called for objects.
- your previous PR only checked for the first element of the array which is in my opinion insufficient as the array could contain objects, but not at the first index.

Though I agree that at least all object keys can and should be sorted, independent of how deeply nested they are.

For that, we shouldn't remove the array check but instead should do something like this:

```js
if (isArray(raw)) {
  return raw.map((raw) => sortDeep(raw))
}
```
That way the arrays are not sorted but any object keys inside these arrays are.

File: api/src/utils/get-snapshot.ts [hunk 1]
```diff
@@ -66,9 +66,5 @@ function sortDeep(raw: any): any {
 		return fromPairs(sorted);
 	}
```
Comment (u12206050) [inline api/src/utils/get-snapshot.ts:68]:
```suggestion
	if (isArray(raw)) {
		return raw.map((raw) => sortDeep(raw))
	}
```

**Comment by u12206050** [comment]
Updated.

**Comment by Nitwel** [comment]
Thanks, if you now could fix the linter errors, I'm happy to merge this in. 😄

**Review by licitdev — COMMENTED** [review]
@Nitwel I think we should add a changeset for this PR 🤔

**Comment by Nitwel** [comment]
Oh, totally forgot that, thanks for reminding me!

**Comment by u12206050** [comment]
So tested this out locally, and unfortunately the export of the the schema in different environments is again giving different outputs, specifically within the `relations` array of objects. Would it be an idea to specifically target and sort the `relations`, `collections` and `fields` array of the schema?

**Comment by rijkvanzanten** [comment]
@u12206050 Sorting the relations by their `id` field, collections alphabetically, and fields by their `sort` value sounds like a reasonable thing to do here? 🤔

**Comment by Nitwel** [comment]
Maybe I'm missing something, but this should be done by the `sortBy` in these lines:
https://github.com/directus/directus/pull/23598/files#diff-c783b2074f310fd793d1653b823996330e13b500204316918a8676a873c51841L32-L40

**Comment by rijkvanzanten** [comment]
@Nitwel Ah! copy that 👍🏻

**Pull request merged**